Formula/kat.rb

```diff
@@ -25,10 +26,16 @@ class Kat < Formula
   depends_on "automake" => :build
   depends_on "libtool" => :build
 
+  depends_on "boost"
```

Comment:
Does `brew linkage kat` report any run time dependency on any `boost` shared library? If not, this can be a `:build` dependency.

Reply:
output of `brew linkage kat`:
```
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /usr/local/opt/boost/lib/libboost_chrono.dylib (boost)
  /usr/local/opt/boost/lib/libboost_filesystem.dylib (boost)
  /usr/local/opt/boost/lib/libboost_program_options.dylib (boost)
  /usr/local/opt/boost/lib/libboost_system.dylib (boost)
  /usr/local/opt/boost/lib/libboost_timer.dylib (boost)
  /usr/local/Cellar/kat/2.4.2/lib/libkat_jellyfish.2.dylib (brewsci/bio/kat)
  /usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/Python (python@3.9)
Indirect dependencies with linkage:
  python@3.9
Dependencies with no linkage:
  brewsci/bio/matplotlib
  scipy
```
@sjackman does this mean we should leave `boost` as-is here?

Reply:
Yes! If you saw `boost` under `Dependencies with no linkage`, then you could change it to
```ruby
  depends_on "boost" => :build
```
I asked specifically about Boost, because a lot of its libraries are only headers only, and have no run time linkage.

`matplotlib` and `scipy` are different because `kat` uses their Python files, and this warning looks only for dynamic library linkage (`.so` on Linux and `.dylib` on macOS).